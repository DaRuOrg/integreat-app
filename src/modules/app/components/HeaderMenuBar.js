import React from 'react'
import PropTypes from 'prop-types'
import cx from 'classnames'
import { isEmpty } from 'lodash/lang'
import { Link } from 'redux-little-router'

import style from './HeaderMenuBar.css'

export const MENU_ITEMS_PROP_TYPE = PropTypes.arrayOf(PropTypes.shape({
  text: PropTypes.string.isRequired,
  href: PropTypes.string.isRequired,
  active: PropTypes.bool.isRequired
}))

class HeaderMenuBar extends React.Component {
  static propTypes = {
    className: PropTypes.string,
    items: MENU_ITEMS_PROP_TYPE.isRequired
  }

  render () {
    return <div className={cx(this.props.className, style.menuBar, isEmpty(this.props.items) ? style.hidden : '')}>
      {this.props.items.map(({href, active, text}) => (
        <Link key={text}
              className={cx(style.menuItem, active ? style.activeMenuItem : '')}
              href={href}>{text}</Link>
      ))}
    </div>
  }
}

export default HeaderMenuBar
