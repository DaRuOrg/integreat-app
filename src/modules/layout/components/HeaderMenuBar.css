@import url('../../common/constants/dimensions.css');
@import url('../../common/constants/helpers.css');

.menu-bar {
  display: flex;
  align-items: center;
  justify-content: center;
}

.menu-bar.hidden {
  display: none;
}

.menu-item {
  @mixin remove-a;
  flex: 1;
  font-size: 1.2em;
  font-weight: 500;
  text-align: center;
}

.menu-item.active-menu-item {
  font-weight: 700;
}

@media (--small-viewport) {
  .menu-item {
    font-size: 1em;
  }
}
