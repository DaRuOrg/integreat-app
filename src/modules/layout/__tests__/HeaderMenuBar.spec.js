import React from 'react'
import { shallow } from 'enzyme'
import HeaderMenuBar from '../components/HeaderMenuBar'

describe('HeaderMenuBar', () => {
  test('should match snapshot', () => {
    const component = shallow(
      <HeaderMenuBar className='testClass' items={[
        {href: 'link1', text: 'text1'},
        {href: 'link2', text: 'text2'}
      ]}/>
    )
    expect(component).toMatchSnapshot()
  })
})
