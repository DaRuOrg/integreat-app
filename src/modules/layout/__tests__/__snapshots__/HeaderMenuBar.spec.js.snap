// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderMenuBar should match snapshot 1`] = `
<div
  className="testClass menuBar"
>
  <Connect(Link)
    className="menuItem"
    href="link1"
    key="text1"
  >
    text1
  </Connect(Link)>
  <Connect(Link)
    className="menuItem"
    href="link2"
    key="text2"
  >
    text2
  </Connect(Link)>
</div>
`;
